client/src/partials/inventory/list/list.js

```diff
@@ -128,28 +128,20 @@ function InventoryListController ($translate, Inventory, Notify, uiGridConstants
 
   /** print inventory list */
   function printList() {
-    var params = formaData(vm.gridOptions);
-    return Modal.openPrinterData(params);
-  }
-
-  /**
-   * format data for modal printer page
-   * @fixme: need of server generate pdf because data
-   * for inventory are very huge for the client
-   */
-  function formaData(gridOptions) {
-    var headers = {};
-    var rows = gridOptions.data;
-
-    gridOptions.columnDefs.forEach(function (col) {
-      headers[col.field] = col.displayName;
-    });
-
-    return {
-      title : 'INVENTORY.LIST',
-      headers : headers,
-      rows : rows
-    };
+    var buttonState = JSON.parse(JSON.stringify(vm.bcButtons[2]));
+    vm.bcButtons[2].disabled = true;
+    vm.bcButtons[2].icon = 'fa fa-spinner fa-pulse';
+
+    Inventory.report('pdf')
+    .then(function (result) {
+      vm.receipt = result;
+      return Modal.openPrinterData({ receipt: vm.receipt, renderer: 'pdf' });
```

Comment:
Let's make a `ReportService` to manage the report modals.  Most modals with be:
 1. Actions (delete, update, accept, etc)
 2. Receipts (patient invoice receipts, voucher receipts, transactions, etc).
 3. Reports (Inventory List, Patient Registrations, etc)

Could you make a special `ReportsService` that render the report, like `ReceiptService` renders the receipt? 

Any future modules will be able to use this service to display their reports.